src/main/java/zaasclient/HttpsClient.java

```diff
@@ -0,0 +1,201 @@
+            File trustFile = new File(trustStorePath);
+            trustStore.load(new FileInputStream(trustFile), trustStorePassword.toCharArray());
+            tmf.init(trustStore);
+        }
+
+        if (keyStorePath != null) {
+            kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+            File keyFile = new File(keyStorePath);
+            keyStore.load(new FileInputStream(keyFile), keyStorePassword.toCharArray());
+            kmf.init(keyStore, keyStorePassword.toCharArray());
+        }
+
+        sslContext = this.getSSLContext(kmf, tmf);
+
+        return getCloseableHttpClient(sslContext);
+    }
+
+    private CloseableHttpClient getCloseableHttpClient(SSLContext sslContext) {
+        final SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslContext,
+            SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+
+        closeableHttpsClient = HttpClients.custom()
+            .setSSLSocketFactory(sslsf)
+            .setDefaultRequestConfig(this.requestConfig)
+            .setMaxConnTotal(3 * 3)
+            .setMaxConnPerRoute(3)
+            .build();
+        return closeableHttpsClient;
+    }
+
+    /**
+     * Overloaded method of the getclosableHttpClient method .. so that the cookies can be used for storing jwt tokens
+     *
+     * @param sslContext
+     * @param cookieStore
+     * @return
+     */
+    private CloseableHttpClient getCloseableHttpClient(SSLContext sslContext, BasicCookieStore cookieStore) {
+        final SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslContext,
+            SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+
+        closeableHttpsClient = HttpClients.custom()
+            .setSSLSocketFactory(sslsf)
+            .setDefaultRequestConfig(this.requestConfig)
+            .setMaxConnTotal(3 * 3)
+            .setMaxConnPerRoute(3)
+            .setDefaultCookieStore(cookieStore)
+            .build();
+        return closeableHttpsClient;
+    }
+
+
+    /**
+     * @param kmf
+     * @param tmf
+     * @return
+     * @throws Exception
+     */
+    private SSLContext getSSLContext(KeyManagerFactory kmf, TrustManagerFactory tmf) throws KeyManagementException, NoSuchAlgorithmException {
+        SSLContext sslContext;
+        try {
+            sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(kmf != null ? kmf.getKeyManagers() : null, tmf != null ? tmf.getTrustManagers() : null, new SecureRandom());
+        } catch (NoSuchAlgorithmException | KeyManagementException e) {
+            throw e;
+        }
+        return sslContext;
+    }
+
+    /**
+     * This method is sed to create custom request configuration
+     *
+     * @return
+     */
+    private RequestConfig buildCustomRequestConfig() {
+        final RequestConfig.Builder builder = RequestConfig.custom();
```

Comment:
It can be rewritten smth like that
```suggestion
        return RequestConfig.custom()
                .setConnectionRequestTimeout(10 * 1000)
                .setSocketTimeout(10 * 1000)
                .setConnectTimeout(10 * 1000)
                .build()
```